alf/utils/value_ops.py

```diff
@@ -187,18 +182,17 @@ def one_step_discounted_return(rewards, values, step_types, discounts):
         A tensor with shape [T-1, B] (or [T-1]) representing the discounted
         returns.
     """
-    assert values.shape[0] >= 2, "The sequence length needs to be \
-                                at least 2. Got {s}".format(s=values.shape[0])
-
-    discounts = discounts[1:]
-    rewards = rewards[1:]
-    values = values[1:]
-    step_types = step_types[:-1]
+    assert values.shape[0] >= 2, ("The sequence length needs to be "
+                                  "at least 2. Got {s}".format(
+                                      s=values.shape[0]))
 
     is_lasts = (step_types == StepType.LAST).to(dtype=torch.float32)
-    returns = rewards + (1 - is_lasts) * discounts * values
 
-    return returns
+    # the following computation can be further simplified but is kept this way
+    # for better understanding
+    rets = (1 - is_lasts[:-1]) * rewards[1:] + discounts[1:] * values[1:] + \
```

Comment:
I think we can directly do `rets=rewards[1:] + discounts[1:] * values[1:]` and remove `is_lasts`

Reply:
Added documentation on this for better understanding: 
``` return = next_reward + next_discount * next_value if is not the last step
    otherwise will set return = next_discount * next_value.```